salesgpt/salesgptapi.py

```diff
@@ -33,42 +32,90 @@ def do(self, conversation_history: [str], human_input=None):
                 )
             else:
                 sales_agent = SalesGPT.from_llm(self.llm, verbose=self.verbose)
-
         else:
             with open(self.config_path, "r") as f:
                 config = json.load(f)
             if self.verbose:
                 print(f"Agent config {config}")
+            if self.use_tools:
+                print("USING TOOLS")
+                config["use_tools"] = True
+                config["product_catalog"] = "examples/sample_product_catalog.txt"
+            else:
+                config.pop("use_tools", None)  # Remove the use_tools key from config if it exists
             sales_agent = SalesGPT.from_llm(self.llm, verbose=self.verbose, **config)
+        print(f"SalesGPT use_tools: {sales_agent.use_tools}")  # Print the use_tools value of the SalesGPT instance
+        sales_agent.seed_agent()
+        return sales_agent
 
-        #  check turns
-        current_turns = len(conversation_history) + 1
+    def do(self, human_input=None):
+        self.current_turn+=1
+        current_turns = self.current_turn
         if current_turns >= self.max_num_turns:
-            # todo:
-            # if self.verbose:
             print("Maximum number of turns reached - ending the conversation.")
-            return "<END_OF_>"
+            return ["BOT","In case you'll have any questions - just text me one more time!"]
 
-        # seed
-        sales_agent.seed_agent()
-        sales_agent.conversation_history = conversation_history
+        #self.sales_agent.seed_agent() why do we seeding at each turn? put to agent_init
+        #self.sales_agent.conversation_history = conversation_history
 
         if human_input is not None:
-            sales_agent.human_step(human_input)
-            # sales_agent.determine_conversation_stage()
-            # print('=' * 10)
-            # print(f"conversation_stage_id:{sales_agent.conversation_stage_id}")
+            self.sales_agent.human_step(human_input)
 
-        sales_agent.step()
-
-        # end conversation
-        if "<END_OF_CALL>" in sales_agent.conversation_history[-1]:
+        ai_log = self.sales_agent.step(stream=False)
+        self.sales_agent.determine_conversation_stage()
+        if "<END_OF_CALL>" in self.sales_agent.conversation_history[-1]:
             print("Sales Agent determined it is time to end the conversation.")
-            return "<END_OF_CALL>"
+            return ["BOT","In case you'll have any questions - just text me one more time!"]
 
-        reply = sales_agent.conversation_history[-1]
+        reply = self.sales_agent.conversation_history[-1]
 
         if self.verbose:
             print("=" * 10)
-            print(f"{sales_agent.salesperson_name}:{reply}")
-        return reply.split(": ")
+        ''''''
+        if ai_log['intermediate_steps'][1]['outputs']['intermediate_steps'] is not []:
```

Comment:
@iljamak this looks convoluted - can you separate it into its own function? And shouldn't it be a part of `parser.py`?